app/mailers/patient_mailer.rb

```diff
@@ -72,9 +73,17 @@ def assessment_sms_weblink(patient)
         add_success_history(dependent, patient)
       else
         add_fail_history_sms(dependent)
+        failed_dependents = true
         patient.update(last_assessment_reminder_sent: nil) # Reset send attempt timestamp on failure
       end
     end
+    return unless patient.dependents.count >= 2 && !patient.active_dependents.include?(patient)
+
+    if failed_dependents
+      add_fail_history_dependents(patient)
+    else
+      add_success_history_dependents(patient)
+    end
   end
 
   def assessment_sms(patient)
```

Comment:
It looks like you only added this for weblinks - but it should be added for all options such as `assessment_sms` and `assessment_voice`. `assessment_email` needs to be updated anyway to include a failure history, which currently is a separate task (https://tracker.codev.mitre.org/browse/SARAALERT-1188). It may make sense to just include that here given we have the language pretty much decided already and it touches the same place.

Reply:
I'd be happy to do 1188 and throw it in this PR. For the other options, I think I might be confused about this bug: looking at the code, the only one of the options where we actually send reminders to each of the relevant patient's active dependents. In assessment_sms and assessment_voice, we send the reminder only to the patient themselves and send it whether or not monitoring is true, so histories would still be created for them in the regular flow.

Reply:
Correct, sms weblinks are the only one where a separate text/weblink is sent for each active dependent. So the additional update you'd need to add is for emailed weblinks as it is now. However, looking at this it seems like there may be more to this bug because for options such as `assessment_sms` we don't have history items on the dependent records at all, which I believe we should to be consistent with weblinks (I thought this was existing functionality, hence my original comment). @mayerm94 Was there was decided upon reason for this that you may be aware of or should this also be resolved?